waku/node/v2/wakunode2.nim

```diff
@@ -47,6 +47,34 @@ proc initAddress(T: type MultiAddress, str: string): T =
     raise newException(ValueError,
                        "Invalid bootstrap node multi-address")
 
+proc removeContentFilters(filters: var Filters, contentFilters: seq[ContentFilter]) {.gcsafe.} =
+  # Flatten all unsubscribe topics into single seq
+  var unsubscribeTopics: seq[ContentTopic]
+  for cf in contentFilters:
+    unsubscribeTopics = unsubscribeTopics.concat(cf.topics)
+  
+  debug "unsubscribing", unsubscribeTopics=unsubscribeTopics
+
+  var rIdToRemove: seq[string] = @[]
+  for rId, f in filters.mpairs:
+    # Iterate filter entries to remove matching content topics
+    for cf in f.contentFilters.mitems:
+      # Iterate content filters in filter entry
+      cf.topics.keepIf(proc (t: auto): bool = t notin unsubscribeTopics)
+    # make sure we delete the content filter
+    # if no more topics are left
+    f.contentFilters.keepIf(proc (cf: auto): bool = cf.topics.len > 0)
```

Comment:
Not that it matters much, but I think here you could do something like:

```nim
f.contentFilters.keepItIf(it.cf.topics notin unsubscribeTopics)
```

not sure tho

Reply:
I think we'll need separate topic iteration, as the `notin` operator for a sequence of items can only be performed one item at a time. But perhaps this will also only modify a content filter if _all_ its content topics are unsubscribed (as opposed to doing so on an individual basis)?